stakeapi/client.py

```
@@ -279,56 +305,60 @@ async def _request(
                 except (json.JSONDecodeError, aiohttp.ContentTypeError, ValueError):
                     body_preview = (await response.text())[:200]
                     raise NetworkError(
-                        f"Expected JSON but got a non-JSON response (status {response.status}) "
-                        f"from {url}. This usually means a Cloudflare challenge page or a "
-                        f"redirect to a login page. Response starts with: {body_preview!r}"
+                        f"Expected JSON but got a non-JSON response "
+                        f"(status {response.status}) from {url}. This usually "
+                        "means a Cloudflare challenge page or a redirect to a "
+                        f"login page. Response starts with: {body_preview!r}"
                     )
```

Ruff (0.16.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)